ambari-server/src/main/java/org/apache/ambari/server/audit/event/LoginAuditEvent.java

```diff
@@ -62,15 +62,14 @@ private LoginAuditEventBuilder() {
     protected void buildAuditMessage(StringBuilder builder) {
       super.buildAuditMessage(builder);
 
-      builder.append(", Operation(User login), Roles(").append(System.lineSeparator());
+      builder.append(", Operation(User login), Roles(");
 
       if (roles != null && !roles.isEmpty()) {
         List<String> lines = new LinkedList<>();
         for (Map.Entry<String, List<String>> entry : roles.entrySet()) {
-          lines.add("    " + entry.getKey() + ": " + StringUtils.join(entry.getValue(), ", "));
+          lines.add("(" +entry.getKey() + ": " + StringUtils.join(entry.getValue(), ", ") + ")");
         }
-        builder.append(StringUtils.join(lines, System.lineSeparator()));
-        builder.append(System.lineSeparator());
+        builder.append(StringUtils.join(lines, ")"));
```

Comment:
I think `)` as separator between roles would be pretty confusing.

Reply:
@adoroszlai in line 72 it is not a separator between the roles . it is only closing the opened bracket.

this is how the actual audit log entry would look alike,
`2019-04-26T17:06:17.451Z, User(test), RemoteIp(10.22.8.5), Operation(User login), Roles(c2111: Cluster Administrator ,Files View: View User ,YARN Queue Manager: View User), Status(Success)`

Reply:
No, the opening bracket is closed in line 74.  Here's an example with multiple roles:

```
2019-04-26T17:35:57.685Z, User(test), RemoteIp(172.27.89.142), Operation(User login), Roles((Files View: View User))(cl1: Service Administrator)), Status(Success)
```

Reply:
thanks for pointing out @adoroszlai - I have corrected that . latest logs shows like this

`2019-04-26T21:08:18.862Z, User(test), RemoteIp(10.22.8.5), Operation(User login), Roles(c2111: Cluster Administrator,Files View: View User,YARN Queue Manager: View User), Status(Success)`
`2019-04-26T21:08:36.964Z, User(admin), RemoteIp(10.22.8.5), Operation(User login), Roles(Ambari: Ambari Administrator), Status(Success)`

Reply:
@adoroszlai  @apappu  if this Conversation is resolved , will you mark as resolve conversation